### PR TITLE
allow trace filters on handler names

### PR DESCRIPTION
**Pull request by harshavardhana**
also convert the filter options to slice flags, for example
if you wish to trace only for status code 404 and 503 then
you would specify the following command

```
mc admin trace --status-code 404 --status-code 503 myminio/
```

Refactor and make implementation more readable.

### CODE REVIEW

**Review by poornas — APPROVED** [review]
LGTM

**Comment by poornas** [comment]
Can update the example to show multiple status-codes can be specified
>mc admin trace --status-codes 404 --status-codes 503 myminio/

**Comment by harshavardhana** [comment]
> Can update the example to show multiple status-codes can be specified
> 
> > mc admin trace --status-codes 404 --status-codes 503 myminio/

done @poornas

File: cmd/admin-trace.go [hunk 1]
```diff
@@ -42,23 +42,27 @@ var adminTraceFlags = []cli.Flag{
 	},
 	cli.BoolFlag{
 		Name:  "all, a",
-		Usage: "trace all traffic (including internode traffic between MinIO servers)",
+		Usage: "trace all traffic (includes internode traffic from MinIO cluster)",
 	},
-	cli.StringFlag{
+	cli.IntSliceFlag{
 		Name:  "status-codes",
```
Comment (vadmeste) [inline cmd/admin-trace.go:48]: Maybe it makes sense to call this 'status-code' instead of 'status-codes', same for other applicable params

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/admin-trace.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/admin-trace.go:48]: done @vadmeste

File: cmd/admin-trace.go [hunk 2]
```diff
@@ -80,18 +84,20 @@ FLAGS:
...
+  1. Show verbose console trace for MinIO server
+     {{.Prompt}} {{.HelpName}} -v -a myminio
 
-  2. Show trace only for failed requests for a MinIO server with alias 'myminio'
+  2. Show trace only for failed requests for MinIO server
     {{.Prompt}} {{.HelpName}} -v -e myminio
 
-  3. Show the content of 503 failed requests
+  3. Show verbose console trace for requests with '503' status code
     {{.Prompt}} {{.HelpName}} -v --status-codes 503 myminio
```
Comment (vadmeste) [inline cmd/admin-trace.go:94]: s/status-codes/status-code/

File: cmd/admin-trace.go [hunk 3]
```diff
@@ -80,18 +84,20 @@ FLAGS:
...
 
-  4. Show all requests for a specific prefix
+  4. Show console trace for a specific path
     {{.Prompt}} {{.HelpName}} --path my-bucket/my-prefix/ myminio
 
+  5. Show console trace for requests with '404' and '503' status code
+    {{.Prompt}} {{.HelpName}} --status-codes 503 myminio
```
Comment (vadmeste) [inline cmd/admin-trace.go:100]: The command does not comply with the example description (you need `--status-code 503 --status-code 404`)

**Review by vadmeste — CHANGES_REQUESTED** [review]
Few comments

**Comment by kannappanr** [comment]
@vadmeste PTAL

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**